### PR TITLE
Extract _unscale_grads helper in LossScaleOptimizer

### PR DESCRIPTION
**Pull request by MarcosAsh**
## Description

`LossScaleOptimizer` unscales gradients in two places, `_stateless_handle_finite_grads` and `_stateful_handle_finite_grads`, and both use the same list comprehension to divide by the loss scale while leaving untouched any gradient that overwrites its variable. This pulls that block into a `_unscale_grads` helper so the rule lives in one place and the two paths cannot drift apart.

There is no behavior change, the two comprehensions were already identical. The helper reads `self.dynamic_scale` itself rather than taking it as an argument, since both call sites were computing that local only to pass it straight through, and in the stateless path the read still happens inside the same `StatelessScope` as before.

Splitting this out of #23076, where @hertschuh noted that the refactor is good but unrelated to MLX and asked for it as a separate PR against master.

Tested with `keras/src/optimizers/loss_scale_optimizer_test.py` on the jax, tensorflow and torch backends.

## Contributor Agreement

**Please review our [AI-Assisted Contribution Policy](../CONTRIBUTING.md#ai-assisted-contribution-policy) and check all boxes below before submitting your PR for review:**

- [x] I am a human, and not a bot.
- [x] I will be responsible for responding to review comments in a timely manner.
- [x] I will work with the maintainers to push this PR forward until submission.

### CODE REVIEW

File: keras/src/optimizers/loss_scale_optimizer.py [hunk 1]
```diff
@@ -141,6 +141,17 @@ def stateless_apply(self, optimizer_variables, grads, trainable_variables):
             ),
         )
 
+    def _unscale_grads(self, grads, variables):
+        # Divide gradients by the loss scale, keeping any gradient that
+        # overwrites its variable untouched.
+        scale = self.dynamic_scale
```
Comment (jeffcarp) [inline keras/src/optimizers/loss_scale_optimizer.py]: It seems like this assignment isn't necessary since it's only used once and isn't at risk of going over the char line limit?

**Review by jeffcarp — COMMENTED** [review]
Thanks! Left one nit before merging

File: keras/src/optimizers/loss_scale_optimizer.py (hunk 1 above)
Comment (MarcosAsh) [reply keras/src/optimizers/loss_scale_optimizer.py]: Fixed this.

**Comment by MarcosAsh** [comment]
Thanks for the review! I fixed the nit.

**Review by jeffcarp — APPROVED** [review]
Thank you!
